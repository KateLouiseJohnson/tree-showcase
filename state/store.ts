import { configureStore } from '@reduxjs/toolkit'
import reducer from './treeShowcaseSlice'

export default configureStore({
  reducer: {
    treeShowcase: reducer,
  }
})