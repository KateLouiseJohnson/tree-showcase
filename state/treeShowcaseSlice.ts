import { createSlice } from '@reduxjs/toolkit'
import { TreeShowcaseState } from './types'

export const initialState: TreeShowcaseState = {
  fetching: false,
  success: false,
  failure: false,
  trees: [],
}

const treesSlice = createSlice({
  name: 'treeShowcase',
  initialState: initialState,
  reducers: {
    setTreesFetching(state) {
      state.success = false
      state.failure = false
      state.fetching = true
    },
    setTreesFetchedSuccess(state, action) {
      state.trees = action.payload
      state.success = true
      state.failure = false
      state.fetching = false
    },
    setTreesFetchedFailure(state) {
      state.success = false
      state.failure = true
      state.fetching = false
    }
  }
})

export const { setTreesFetching, setTreesFetchedSuccess, setTreesFetchedFailure } = treesSlice.actions
export default treesSlice.reducer
