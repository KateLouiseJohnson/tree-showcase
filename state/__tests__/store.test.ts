import {describe, expect, it} from '@jest/globals'
import store from '../store'
import { serialisedTrees } from '../../fixtures'
import { setTreesFetching, setTreesFetchedSuccess, setTreesFetchedFailure } from '../treeShowcaseSlice'

describe('Tree showcase state test', () => {
  it('should set initial state', async () => {
    const state = store.getState()
    expect(state).toMatchSnapshot()
  })

  it('should set loading', async () => {
    const state = store.dispatch(setTreesFetching())
    expect(state).toMatchSnapshot()
    expect(store.getState()).toMatchSnapshot()
  })

  it('should set failed', async () => {
    const state = store.dispatch(setTreesFetchedFailure())
    expect(state).toMatchSnapshot()
    expect(store.getState()).toMatchSnapshot()
  })

  it('should set success', async () => {
    const state = store.dispatch(setTreesFetchedSuccess(serialisedTrees))
    expect(state).toMatchSnapshot()
    expect(store.getState()).toMatchSnapshot()
  })
})