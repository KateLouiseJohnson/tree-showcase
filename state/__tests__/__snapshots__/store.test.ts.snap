// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree showcase state test should set failed 1`] = `
Object {
  "payload": undefined,
  "type": "treeShowcase/setTreesFetchedFailure",
}
`;

exports[`Tree showcase state test should set failed 2`] = `
Object {
  "treeShowcase": Object {
    "failure": true,
    "fetching": false,
    "success": false,
    "trees": Array [],
  },
}
`;

exports[`Tree showcase state test should set initial state 1`] = `
Object {
  "treeShowcase": Object {
    "failure": false,
    "fetching": false,
    "success": false,
    "trees": Array [],
  },
}
`;

exports[`Tree showcase state test should set loading 1`] = `
Object {
  "payload": undefined,
  "type": "treeShowcase/setTreesFetching",
}
`;

exports[`Tree showcase state test should set loading 2`] = `
Object {
  "treeShowcase": Object {
    "failure": false,
    "fetching": true,
    "success": false,
    "trees": Array [],
  },
}
`;

exports[`Tree showcase state test should set success 1`] = `
Object {
  "payload": Object {
    "trees": Array [
      Object {
        "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/3/36/Baobab_Adansonia_digitata.jpg",
        "name": "Baobab",
        "species": "Adansonia",
      },
      Object {
        "imageUrl": "https://upload.wikimedia.org/wikipedia/en/1/16/Red_mangrove-everglades_natl_park.jpg",
        "name": "Red Mangrove",
        "species": "Rhizophora mangle",
      },
      Object {
        "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/2/2c/Carpinus_betulus_-_Hunsr%C3%BCck_001.jpg",
        "name": "Common Hornbeam",
        "species": "Carpinus betulus",
      },
      Object {
        "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/3/34/Quercus_cerris.JPG",
        "name": "Turkey Oak",
        "species": "Quercus cerris",
      },
      Object {
        "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/f/f0/Pinus_syluestriformis_%28Takenouchi%29T.Wang_ex_Cheng.JPG",
        "name": "Japanese red pine",
        "species": "Pinus densiflora",
      },
    ],
  },
  "type": "treeShowcase/setTreesFetchedSuccess",
}
`;

exports[`Tree showcase state test should set success 2`] = `
Object {
  "treeShowcase": Object {
    "failure": false,
    "fetching": false,
    "success": true,
    "trees": Object {
      "trees": Array [
        Object {
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/3/36/Baobab_Adansonia_digitata.jpg",
          "name": "Baobab",
          "species": "Adansonia",
        },
        Object {
          "imageUrl": "https://upload.wikimedia.org/wikipedia/en/1/16/Red_mangrove-everglades_natl_park.jpg",
          "name": "Red Mangrove",
          "species": "Rhizophora mangle",
        },
        Object {
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/2/2c/Carpinus_betulus_-_Hunsr%C3%BCck_001.jpg",
          "name": "Common Hornbeam",
          "species": "Carpinus betulus",
        },
        Object {
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/3/34/Quercus_cerris.JPG",
          "name": "Turkey Oak",
          "species": "Quercus cerris",
        },
        Object {
          "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/f/f0/Pinus_syluestriformis_%28Takenouchi%29T.Wang_ex_Cheng.JPG",
          "name": "Japanese red pine",
          "species": "Pinus densiflora",
        },
      ],
    },
  },
}
`;
